Source/Core/DolphinQt2/Config/Graphics/GraphicsWindow.cpp

```diff
@@ -0,0 +1,141 @@
+#include "DolphinQt2/Config/Graphics/EnhancementsWidget.h"
+#include "DolphinQt2/Config/Graphics/GeneralWidget.h"
+#include "DolphinQt2/Config/Graphics/HacksWidget.h"
+#include "DolphinQt2/Config/Graphics/SoftwareRendererWidget.h"
+#include "DolphinQt2/MainWindow.h"
+
+GraphicsWindow::GraphicsWindow(X11Utils::XRRConfiguration* xrr_config, MainWindow* parent)
+    : QDialog(parent), m_xrr_config(xrr_config)
+{
+  CreateMainLayout();
+  ConnectWidgets();
+
+  setWindowTitle(tr("Graphics"));
+  setWindowFlags(Qt::Window);
+
+  OnBackendChanged(QString::fromStdString(SConfig::GetInstance().m_strVideoBackend));
+
+  connect(parent, &MainWindow::EmulationStarted, this, &GraphicsWindow::EmulationStarted);
+  connect(parent, &MainWindow::EmulationStopped, this, &GraphicsWindow::EmulationStopped);
+}
+
+void GraphicsWindow::CreateMainLayout()
+{
+  auto* main_layout = new QVBoxLayout();
+  auto* description_box = new QGroupBox(tr("Description"));
+  auto* description_layout = new QVBoxLayout();
+  m_description =
+      new QLabel(tr("Move the mouse pointer over an option to display a detailed description."));
+  m_tab_widget = new QTabWidget();
+  m_button_box = new QDialogButtonBox(QDialogButtonBox::Ok);
+
+  description_box->setLayout(description_layout);
+  description_box->setMinimumHeight(205);
+
+  m_description->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Expanding);
+  m_description->setWordWrap(true);
+  m_description->setAlignment(Qt::AlignTop | Qt::AlignLeft);
+
+  description_layout->addWidget(m_description);
+
+  main_layout->addWidget(m_tab_widget);
+  main_layout->addWidget(description_box);
+  main_layout->addWidget(m_button_box);
+
+  m_general_widget = new GeneralWidget(m_xrr_config, this);
+  m_enhancements_widget = new EnhancementsWidget(this);
+  m_hacks_widget = new HacksWidget(this);
+  m_advanced_widget = new AdvancedWidget(this);
+  m_software_renderer = new SoftwareRendererWidget(this);
+
+  connect(m_general_widget, &GeneralWidget::BackendChanged, this,
+          &GraphicsWindow::OnBackendChanged);
+  connect(m_software_renderer, &SoftwareRendererWidget::BackendChanged, this,
+          &GraphicsWindow::OnBackendChanged);
+
+  if (SConfig::GetInstance().m_strVideoBackend != "Software Renderer")
+  {
+    m_tab_widget->addTab(m_general_widget, tr("General"));
+    m_tab_widget->addTab(m_enhancements_widget, tr("Enhancements"));
+    m_tab_widget->addTab(m_hacks_widget, tr("Hacks"));
+    m_tab_widget->addTab(m_advanced_widget, tr("Advanced"));
+  }
+  else
+  {
+    m_tab_widget->addTab(m_software_renderer, tr("Software Renderer"));
+  }
+
+  setLayout(main_layout);
+}
+
+void GraphicsWindow::ConnectWidgets()
+{
+  connect(m_button_box, &QDialogButtonBox::accepted, this, &QDialog::accept);
+}
+
+void GraphicsWindow::OnBackendChanged(const QString& backend)
+{
+  setWindowTitle(tr("Dolphin %1 Graphics Configuration").arg(backend));
+  if (backend == QStringLiteral("Software Renderer") && m_tab_widget->count() > 1)
```

Comment:
Having this logic depend on the number of tab widgets currently visible seems fragile to me, could you make it depend directly on the old backend? Maybe an `m_current_backend_name`, or something like that?